src/bin/aperf.rs

```diff
@@ -35,14 +40,36 @@ enum Commands {
     Report(Report),
 }
 
-fn init_logger(verbose: u8) -> Result<()> {
+fn init_logger(verbose: u8, runlog: &PathBuf) -> Result<()> {
     let level = match verbose {
         0 => LevelFilter::Info,
         1 => LevelFilter::Debug,
         2 => LevelFilter::Trace,
         _ => return Err(PDError::InvalidVerboseOption.into()),
     };
-    Builder::new().filter_level(level).init();
+    let stdout = ConsoleAppender::builder()
+        .encoder(Box::new(PatternEncoder::new(
+            "[{d(%Y-%m-%dT%H:%M:%SZ)} {h({l}):5.5} {M}] {m}{n}",
+        )))
+        .build();
+
+    let fileout = FileAppender::builder()
+        .encoder(Box::new(PatternEncoder::new(
+            "[{d(%Y-%m-%dT%H:%M:%SZ)} {h({l}):5.5} {M}] {m}{n}",
```

Comment:
Unless you intend the patterns to diverge, factor out the PatternEncoder so it's not written twice.

Reply:
Changed.